### PR TITLE
refactor: consolidate oracle and provider orchestrator

### PR DESCRIPTION
**Pull request by technicallyty**
- folds the logic from orchestrator into Oracle. this removes some of the hairy indirection between the two services. 
- OracleServer no longer starts the oracle. the `main` function starts the oracle, and then passes it into the server

### CODE REVIEW

File: pkg/math/testutils/median.go
Comment (technicallyty) [inline pkg/math/testutils/median.go:42]: the oracle used a general interface of an aggregator called `PriceAggregator`. in the orchestrator, it was explicitly the `IndexAggregator`. 

assuming we want to continue with the generalizable interface, i'm adding this method here. i'm not sure if it needs to do anything though.

File: service/servers/oracle/server_test.go
Comment (technicallyty) [inline service/servers/oracle/server_test.go:241]: i have no idea what this was supposed to do so im just removing it 🤔

File: pkg/math/testutils/median.go
Comment (davidterpay) [reply pkg/math/testutils/median.go:42]: Yea this doesn't need to do anything. We don't use this aggregator and is mostly here just for testing purposes.

File: oracle/helpers_test.go
Comment (davidterpay) [inline oracle/helpers_test.go:217]: Might be useful to move this into the testutils directory.

File: oracle/oracle.go
Comment (davidterpay) [inline oracle/oracle.go]: nit: can you move this and all other interfaces to the `interfaces.go` file.

File: oracle/oracle.go
Comment (davidterpay) [inline oracle/oracle.go:100]: mega nit: would add nil check on the aggregator.

File: oracle/oracle.go
Comment (davidterpay) [inline oracle/oracle.go]: Is there a reason we removed the read lock on this?

File: oracle/lifecycle.go
Comment (davidterpay) [inline oracle/lifecycle.go]: nit: can we add back the godoc here.

File: oracle/market_mapper.go
Comment (davidterpay) [inline oracle/market_mapper.go:17]: We can probably just exit if `mmProvider` is nil.

File: oracle/options.go
Comment (davidterpay) [inline oracle/options.go:88]: Is this still necessary?

File: oracle/options.go
Comment (davidterpay) [inline oracle/options.go]: nit: can you add go docs

File: cmd/slinky/main.go
Comment (davidterpay) [inline cmd/slinky/main.go]:
```suggestion
	// Define the oracle options. These determine how the oracle is created & executed.
```

File: oracle/oracle.go
Comment (davidterpay) [reply oracle/oracle.go]: I think we still need this.

**Comment by davidterpay** [comment]
lgtm, mostly nits. will wait for deployment metrics before we approval.

File: oracle/options.go
Comment (aljo242) [reply oracle/options.go]: Don't all oracle instances need to have a marketmap provider?



I think this should be a required argument rather than an `Option`

File: oracle/market_mapper.go
Comment (technicallyty) [reply oracle/market_mapper.go:17]: this function is only called if `o.mmProvider != nil`. i could still add it though for extra safety in case it gets called elsewhere in the future, but the original code did not check for this.

File: oracle/options.go
Comment (technicallyty) [reply oracle/options.go:88]: so this was to allow mock providers to be passed into the oracle. i don't think we can stub in fake provider responses without this. see `providers_test.go`.

File: oracle/options.go
Comment (technicallyty) [reply oracle/options.go]: actually, i forgot to remove this option. marketmap providers are instantiated via OracleConfig.Providers, so this option is unused now.

File: oracle/oracle.go
Comment (technicallyty) [reply oracle/oracle.go]: done [move interfaces to interface.go](https://github.com/skip-mev/slinky/pull/503/commits/12ba0b0d4f7fca0621b4a5a447631e9f476c61b7)

File: oracle/oracle.go
Comment (technicallyty) [reply oracle/oracle.go:100]: oops. fixed here [add read lock to GetLastSyncTime and nil check aggregator](https://github.com/skip-mev/slinky/pull/503/commits/8cb65abfcb2f0d9f70bfcc91acb30b11a7416198)

File: oracle/oracle.go
Comment (technicallyty) [reply oracle/oracle.go]: kek i dont know why this got removed, added back here [add read lock to GetLastSyncTime and nil check aggregator](https://github.com/skip-mev/slinky/pull/503/commits/8cb65abfcb2f0d9f70bfcc91acb30b11a7416198)

File: oracle/lifecycle.go
Comment (technicallyty) [reply oracle/lifecycle.go]: https://github.com/skip-mev/slinky/pull/503/commits/bf436df5a1b3bf7fd24808bae54a765834ee8c44

File: oracle/options.go
Comment (technicallyty) [reply oracle/options.go]: added godocs to the other options [add godocs to other options](https://github.com/skip-mev/slinky/pull/503/commits/f9c7bd4f4371c2faa5e6a1bb2fc529736dcdded0)

File: oracle/helpers_test.go
Comment (technicallyty) [reply oracle/helpers_test.go:217]: i tend to prefer to keep stuff like this in the package where its used, and only move it out when some other package needs it. happy to move it out if you disagree though

**Review by aljo242 — APPROVED** [review]
lgtm

**Review by davidterpay — APPROVED** [review]

**Pull request merged**
